tests/test_jsonapi/test_fetching_data.py

```diff
@@ -377,6 +377,25 @@ def test_relationship_links(self):
         assert links['self'].endswith('/article/1/relationships/author')
         assert links['related'].endswith('/article/1/author')
 
+    def test_explicitly_set_primary_key_in_links(self):
+        """Should use the primary_key explicitly set when generating links"""
+        self.manager.create_api(self.Article, primary_key='title')
+        article = self.Article(id=1, title=u'my_article')
+        self.session.add(article)
+        self.session.commit()
+        response = self.app.get('/api/article')
+        data = loads(response.data)['data'][0]
```

Comment:
To match the other tests, this should be

    document = loads(data)
    articles = document['data']
    article = articles[0]

After that, you can use variables to avoid the long lines:

    author_links = article['relationships']['author']['links']
    comments_links = article['relationships']['comments']['links']
    ...